### PR TITLE
ISSUE-62: Duplicate server module

### PR DESCRIPTION
**Pull request by LanKhuat**

### CODE REVIEW

**Comment by chibenwa** [comment]
Please do change the docker image name :-p

File: openpaas-james/apps/distributed-es6-backport/docker-compose.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ services:
       - tika
       - rabbitmq
       - s3
-    image: linagora/openpaas-james-distributed
+    image: linagora/openpaas-james-distributed-es6-backport
```
Comment (chibenwa) [inline openpaas-james/apps/distributed-es6-backport/docker-compose.yml:15]: We should use OpenDistro here?

File: openpaas-james/apps/distributed-es6-backport/pom.xml [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+                            <goal>dockerBuild</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <from>
+                        <image>adoptopenjdk:11-jdk-hotspot</image>
+                    </from>
+                    <to>
+                        <image>linagora/openpaas-james-distributed</image>
```
Comment (Arsnael) [inline openpaas-james/apps/distributed-es6-backport/pom.xml:314]: `linagora/openpaas-james-distributed` => `linagora/openpaas-james-distributed-es6-backport` ?

**Comment by Arsnael** [comment]
```
04:28:39.313 [ERROR] o.a.j.GuiceJamesServer - Fatal error while starting James

org.apache.james.StartUpChecksPerformer$StartUpChecksException: StartUpChecks got bad results: CheckResult{name=ElasticSearchStartUpCheck, resultType=BAD, description=Optional[ES version(6.3.2) is not compatible with the recommendation(7.10.2)]}

	at org.apache.james.StartUpChecksPerformer.performCheck(StartUpChecksPerformer.java:128)

	at org.apache.james.GuiceJamesServer.start(GuiceJamesServer.java:92)

	at org.apache.james.JamesServerMain.main(JamesServerMain.java:30)

	at com.linagora.openpaas.james.app.DistributedServer.main(DistributedServer.java:164)

Exception in thread "main" org.apache.james.StartUpChecksPerformer$StartUpChecksException: StartUpChecks got bad results: CheckResult{name=ElasticSearchStartUpCheck, resultType=BAD, description=Optional[ES version(6.3.2) is not compatible with the recommendation(7.10.2)]}

	at org.apache.james.StartUpChecksPerformer.performCheck(StartUpChecksPerformer.java:128)

	at org.apache.james.GuiceJamesServer.start(GuiceJamesServer.java:92)

	at org.apache.james.JamesServerMain.main(JamesServerMain.java:30)

	at com.linagora.openpaas.james.app.DistributedServer.main(DistributedServer.java:164)
```
You likely forgot to update the test suite for es7 that are likely still linked to a docker es 6?

**Comment by Arsnael** [comment]
You miss deployment tests for the es6 backup as well (as stated in the DoD)

**Comment by Arsnael** [comment]
Rebased on master.

Left to do:
* deployment tests for ES6 backport
* Update Jenkinsfile (mainly for docker jib image push I think)

**Comment by Arsnael** [comment]
What do we do for pushing the docker image of the es6 backport? can we start creating our own image repos with tmail now? If we try to push both distributed and distributed es6 backport into the `linagora/james-rabbitmq-project` with different tags it makes the Jenkinsfile stupidly more complex...

File: openpaas-james/apps/distributed-es6-backport/src/main/conf/elasticsearch.properties [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+elasticsearch.nb.shards=5
+elasticsearch.nb.replica=1
+elasticsearch.index.waitForActiveShards=1
+elasticsearch.retryConnection.maxRetries=7
+elasticsearch.retryConnection.minDelay=3000
+# Index or not attachments (default value: true)
+elasticsearch.indexAttachments=true
+
+# Reports for metrics into ElasticSearch
```
Comment (chibenwa) [inline openpaas-james/apps/distributed-es6-backport/src/main/conf/elasticsearch.properties:77]: Unrelated but we can drop the metric part of this file in TMail as we do not rely on the ElasticSearch metric reporter...

File: openpaas-james/apps/distributed-es6-backport/src/main/conf/jmap.properties [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# Configuration file for JMAP
+# Read https://james.apache.org/server/config-jmap.html for further details
+
+enabled=true
+
+tls.keystoreURL=file://conf/keystore
+tls.secret=james72laBalle
```
Comment (chibenwa) [inline openpaas-james/apps/distributed-es6-backport/src/main/conf/jmap.properties:7]: We don't need JMAP TLS config

**Review by chibenwa — APPROVED** [review]
Looks good to me.

File: openpaas-james/apps/distributed-es6-backport/src/main/conf/jmap.properties (hunk 4 above)
Comment (Arsnael) [reply openpaas-james/apps/distributed-es6-backport/src/main/conf/jmap.properties:7]: Why not?

Comment (Arsnael) [reply openpaas-james/apps/distributed-es6-backport/src/main/conf/jmap.properties:7]: Need this for keystore JWT generation

Comment (chibenwa) [reply openpaas-james/apps/distributed-es6-backport/src/main/conf/jmap.properties:7]: A longer answer is that it is needed by JMAP draft to sign continuation tokens. Legacy...

**Review by chibenwa — APPROVED** [review]

**Comment by Arsnael** [comment]
> What do we do for pushing the docker image of the es6 backport? can we start creating our own image repos with tmail now? If we try to push both distributed and distributed es6 backport into the `linagora/james-rabbitmq-project` with different tags it makes the Jenkinsfile stupidly more complex...

WDYT on this @chibenwa btw?

**Comment by chibenwa** [comment]
Not strongly opinionated to be honnest.

I should check my permissions tomorrow, I likely have them.

As long as we keep in mind that tmail might change I have no issue using distincts dockerhub repos.

**Comment by Arsnael** [comment]
I think I have the permissions now too. Just asking for the course of action :)

**Comment by Arsnael** [comment]
Will change the Jenkinsfile in an other PR

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**